contracts/Gateway.sol

```diff
@@ -17,15 +17,17 @@ contract Gateway is
 {
     NativeTokenPredicate public nativeTokenPredicate;
     IValidators public validators;
+    uint256 public feeAmount;
 
     /// @custom:oz-upgrades-unsafe-allow constructor
     constructor() {
         _disableInitializers();
     }
 
-    function initialize() public initializer {
+    function initialize(uint256 _feeAmount) public initializer {
```

Comment:
rename `feeAmount` to `minFeeAmount`
why we do not have `minAmount` also ? (for `receivers[i].amount` check)